### PR TITLE
Add cleanup rules to Makefile

### PR DESCRIPTION
**Pull request by gkurz**
This provides a finer grained way to prune generated files from the current tree over calling `git clean`. Each target that generates files gets an associated `${target}-clean` : this provides a clear view of which target generates what. Only exceptions are downloaded binaries which all get pruned by a single `bin-clean` target for the sake of simplicity.

Signed-off-by: Greg Kurz <groug@kaod.org>

**- Description of the problem which is fixed/What is the use case**

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: quay.io/rhgkurz/openshift-sandboxed-containers-operator
```
Comment (littlejawa) [inline config/manager/kustomization.yaml:15]: I think this is unrelated to this PR ?

Comment (gkurz) [reply config/manager/kustomization.yaml:15]: > I think this is unrelated to this PR ?

My bad, I forgot to `git reset --hard` _again_ before pushing... I hate `go` for this. :-\

Comment (gkurz) [reply config/manager/kustomization.yaml]: Fixed @littlejawa .

**Review by littlejawa — APPROVED** [review]
LGTM

Thanks !

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @gkurz

**Review by cpmeadors — APPROVED** [review]
Sweet changes!  I love make clean.  I was manually doing this.  Looks good.

**Comment by cpmeadors** [comment]
Prow checks will not pass currently.  use {slash}override <checkname> to overcome this.

**Comment by gkurz** [comment]
/override ci/prow/4.9-ci-index-sandboxed-containers-operator-bundle ci/prow/4.9-images ci/prow/4.9-sandboxed-containers-operator-e2e ci/prow/ci-index-sandboxed-containers-operator-bundle ci/prow/images ci/prow/sandboxed-containers-operator-e2e pull-ci-openshift-sandboxed-containers-operator-master-4.9-ci-index-sandboxed-containers-operator-bundle

**Review by jensfr — APPROVED** [review]
lgtm, thank you @bpradipt

**Comment by jensfr** [comment]
/override ci/prow/4.9-ci-index-sandboxed-containers-operator-bundle ci/prow/4.9-images ci/prow/4.9-sandboxed-containers-operator-e2e ci/prow/ci-index-sandboxed-containers-operator-bundle ci/prow/images ci/prow/sandboxed-containers-operator-e2e pull-ci-openshift-sandboxed-containers-operator-master-4.9-ci-index-sandboxed-containers-operator-bundle
